plugin/awesoMD/plugin.js

```diff
@@ -454,6 +462,34 @@ const Plugin = () => {
 		return [content, options];
 	}
 
+	/**
+	 * Add slide separator in case where the heading indicates the start of new slide
+	 *
+	 * Returns the updated markdown file with added slide separator above every slide headings
+	 */
+	function addSlideSeparator(markdown) {
+		const defaultSlideSeparator = '---';
+		const lines = markdown.split('\n');
+		const result = [];
+		let isFirstHeading = false;
+
+		lines.forEach((line, index) => {
+			if (line.match(/^#+\s/)) {
+				if (!isFirstHeading) {
+					isFirstHeading = true;
```

Comment:
with that logic for the first heading the value would be `false` and for the rest of the execution `true`, that is fine but does not fit with the name of the variable.
So ether change the name of the variable to something like `firstHeadingProcessingDone` or swap `true` and `false` here and in the variable declaration